### PR TITLE
Update Cargo.lock

### PR DESCRIPTION
**Pull request by Mark-Simulacrum**
This also includes major version bumps for the rand crate used by core, std, and alloc tests, among other crates (regex, etc.) used elsewhere. Since these are all internal there should be no user-visible changes.

r? @alexcrichton

### CODE REVIEW

**Comment by alexcrichton** [comment]
Looks good to me! r=me with travis passing

File: src/librustc_cratesio_shim/Cargo.toml [hunk 1]
```diff
@@ -22,3 +22,4 @@ crate-type = ["dylib"]
 [dependencies]
 bitflags = "1.0"
 log = "0.4"
+unicode_width = "0.1.4"
```
Comment (kennytm) [inline src/librustc_cratesio_shim/Cargo.toml]: (should be `unicode-width`)

**Comment by Mark-Simulacrum** [comment]
Okay, I guess adding unicode_width the the crates.io shim isn't quite enough. I'll try to find some time later today to investigate why.

**Comment by Mark-Simulacrum** [comment]
@alexcrichton Okay, I've spent some more time on this and either I just don't understand how rustc works or I'm missing something:

`unicode-width` is depended upon by both syntax_pos and rustc_errors, but that feels like it shouldn't cause any problems, because rustc_errors also depends on syntax_pos. To me, that suggests that rustc should resolve the dependency graph such that unicode width is loaded from syntax_pos's dylib and there'd be no problems.

A slight complication is that `getopts` also depends on `unicode-width`, and I think that might be the source of the problem? Adding getopts and unicode-width the crates io shim crate didn't help though.

It's also not clear to me why we have *both* the crates io shim crate and librustc has a few 'fake' dependencies itself. Shouldn't they serve the same purpose?

**Comment by TimNN** [comment]
Ping from triage @Mark-Simulacrum! Do you think you'll be able to continue to work on this?

**Comment by Mark-Simulacrum** [comment]
I'll get to this soon hopefully.

**Pull request merged**